src/Microsoft.Health.Dicom.Api/Registration/DicomServerServiceCollectionExtensions.cs

```diff
@@ -174,6 +176,9 @@ public static IDicomServerBuilder AddHostedServices(this IDicomServerBuilder ser
         services.AddRecyclableMemoryStreamManager(configurationRoot);
 
         services.AddSingleton<ITelemetryInitializer, TelemetryInitializer>();
+        // track common Dicom server specific requests telemetry
+        services.AddSingleton<ITelemetryInitializer, DicomTelemetryInitializer>();
```

Comment:
I believe you would replace the `ITelemetryInitializer` initialized above since it is a singleton. Do you need the implementation of both?